### PR TITLE
Add Filter setting components.

### PR DESCRIPTION
**Pull request by grafixeyehero**

### CODE REVIEW

**Review by thornbill — CHANGES_REQUESTED** [review]

File: src/apps/experimental/components/library/filter/FiltersYears.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import type { QueryFiltersLegacy } from '@jellyfin/sdk/lib/generated-client';
+import React, { FC, useCallback } from 'react';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import { LibraryViewSettings } from 'types/library';
+
+interface FiltersYearsProps {
+    filtes?: QueryFiltersLegacy;
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/apps/experimental/components/library/filter/FiltersYears.tsx]: https://github.com/jellyfin/jellyfin-web/pull/4711/commits/9cd330ef1580063b03c265b585433be45a948c9c

File: src/apps/experimental/components/library/filter/FiltersTags.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import type { QueryFiltersLegacy } from '@jellyfin/sdk/lib/generated-client';
+import React, { FC, useCallback } from 'react';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import { LibraryViewSettings } from 'types/library';
+
+interface FiltersTagsProps {
+    filtes?: QueryFiltersLegacy;
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/apps/experimental/components/library/filter/FiltersTags.tsx]: https://github.com/jellyfin/jellyfin-web/pull/4711/commits/9cd330ef1580063b03c265b585433be45a948c9c

File: src/apps/experimental/components/library/filter/FiltersGenres.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import type { QueryFiltersLegacy } from '@jellyfin/sdk/lib/generated-client';
+import React, { FC, useCallback } from 'react';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import { LibraryViewSettings } from 'types/library';
+
+interface FiltersGenresProps {
+    filtes?: QueryFiltersLegacy;
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/apps/experimental/components/library/filter/FiltersGenres.tsx]: https://github.com/jellyfin/jellyfin-web/pull/4711/commits/9cd330ef1580063b03c265b585433be45a948c9c

File: src/apps/experimental/components/library/filter/FiltersStudios.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import type { BaseItemDtoQueryResult } from '@jellyfin/sdk/lib/generated-client';
+import React, { FC, useCallback } from 'react';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import { LibraryViewSettings } from 'types/library';
+
+interface FiltersStudiosProps {
+    filtes?: BaseItemDtoQueryResult;
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/apps/experimental/components/library/filter/FiltersStudios.tsx]: https://github.com/jellyfin/jellyfin-web/pull/4711/commits/9cd330ef1580063b03c265b585433be45a948c9c

File: src/hooks/useFetchItems.ts [hunk 5]
```diff
@@ -19,7 +21,7 @@ import { Sections, SectionsViewType } from 'types/suggestionsSections';
 
 const fetchGetItem = async (
     currentApi: JellyfinApiContext,
-    parentId?: string | null,
+    parentId: string | null | undefined,
```
[the remark this answers is not in the document]
Comment (grafixeyehero) [reply src/hooks/useFetchItems.ts]: https://github.com/jellyfin/jellyfin-web/pull/4711/commits/e9fb141431ab095c85da34669e0ee159eeb233a4

**Review by thornbill — APPROVED** [review]

**Pull request merged**